### PR TITLE
Fix compile error for max with inline const

### PR DESCRIPTION
**Pull request by ignaciosica**
Fixes #5838

I think that for correctness this should be generalized for other Ops as well as the same problem may arise in those cases to, but if so many symbolic tests fail as it compares the generated output with fixed ones

Or by directly casting every const, but the same problem arises

### CODE REVIEW

**Comment by Qazalin** [comment]
can you add tests for this? the best file for this is test/test_uops.py.
I wanna see tests for every ALU in code_for_uop, max isn't the only one with this problem.

**Comment by ignaciosica** [comment]
Sure, adding tests today

**Comment by ignaciosica** [comment]
Sorry I couldn't finish it yet, I going to work on it this weekend

**Comment by ignaciosica** [comment]
Sorry I don't have the tests yet, they weren't as straightforward as I expected as const ops are folded before the kernel renders, still working on them though. Maybe there is a straightforward way but I am not seeing it right now

**Comment by Qazalin** [comment]
added `test_renderer_failures.py` - should be enough infra for this diff.

**Comment by ignaciosica** [comment]
Ok, the testing for unary ops was rather tricky as it was constantly being fold before rendering. I could not manage to come up with failing case, sorry

**Comment by Qazalin** [comment]
you don't need to test unary we fold it anyways.

**Comment by ignaciosica** [comment]
> can you add tests for this? the best file for this is test/test_uops.py. I wanna see tests for every ALU in code_for_uop, max isn't the only one with this problem.

MAX is the only BinaryOp in METAL `code_for_uop`, or you meant default `code_for_uop`
image

Do you see ops like `BinaryOps.MUL: lambda a,b,dtype: f"({a}*{b})"` failing because of this issue?

**Comment by Qazalin** [comment]
yea max should be the only one

**Comment by ignaciosica** [comment]
> added `test_renderer_failures.py` - should be enough infra for this diff.

So is this test enough?

**Comment by geohot** [comment]
Tests failing

**Comment by ignaciosica** [comment]
> Tests failing

fixed, no longer expects compiler error

File: tinygrad/renderer/cstyle.py [hunk 1]
```diff
@@ -137,6 +137,7 @@ def ssa(prefix:str, u:Optional[UOp]=None):
         elif uop is UOps.ALU:
           # remove parens if ALU types are the same. TODO: can do more here
           if args in {BinaryOps.ADD,BinaryOps.MUL,BinaryOps.XOR}: operands = [strip_parens(r[v]) if v.arg == args else r[v]for v in src]
+          elif args is BinaryOps.MAX: operands = [self.render_cast(r[v],v.dtype) if v.op is UOps.CONST and v.dtype is not None else r[v] for v in src]
```
Comment (Qazalin) [inline tinygrad/renderer/cstyle.py:140]: you can cast the dtype here with cast(DType
we assert CONST always has a dtype.

Comment (ignaciosica) [reply tinygrad/renderer/cstyle.py]: Sure, changed

**Pull request merged**